module-2/backend/week_one.md

```diff
@@ -5,30 +5,99 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+      Because it allows for easy understanding across all programs. It is saved time that would otherwise be spend figuring  
+      out what extension goes to what page. 
+6. What types of variables are accessible in our view templates without explicitly passing them? 
+      yield
 7. Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template?
   
   ```ruby
-  get '/horses' do
+  get '/horses' do 
     erb :index
   end
   ```
-
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
+    Add => @count = 1 above erb:index 
+    
+8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view? 
+    Add => name = "Mr. Ed"
+9. What's the purpose of ERB? 
+    To allow ruby and html/css to interact together 
+10. Why do I need a development AND test database? 
+    Development is just set/moving things in and around 
+    Test is creating instances that we don't want to stick around after we have tested a thing. 
 11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
+    When you resize a browser window and the site respondse in a way that still makes sense to the user viewing the new size
+12. What is CRUD and why is it important? 
+    Create 
+    Read 
+    Update
+    Delete 
+    
+    These are important because it explains all of the functionality of a website. There isn't anything more that needs to be 
+    done besides C/R/U/D
+13. What does HTTP stand for?  
+    Hyper
+    Text 
+    Transfer
+    Protocol
+14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways? 
+    adding <%= @name %> or #{@name}
+15. What's an ORM? 
+    Object 
+    Relational
+    Mapping
 16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
+    Rails 
+17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for. 
+    1. /restaurants  
+        Get 
+        - gets :index and displays
+    2.  /restaurants/new  
+        Get 
+        - gets :new and displays, user can enter in form
+    3. /restaurants/new  

```

Comment:
The route for this is actually `/restaurants`.

We only use the `/new` to get the form to create a new restaurant.